### DISCUSSION TITLE
Would that be possible to implement Resampling with halo minimization? Looks far superior than unsharp of ImageMagick

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [ImageMagick/ImageMagick#3881] [Help]
There are some scripts posted here but they are not working on a Windows machine (at least I have failed make them work)

http://im.snibgo.com/resamphm.htm#downshp

Resampling with halo minimization looks far superior than unsharp option of ImageMagick

![image](https://user-images.githubusercontent.com/19240467/123564007-54b87700-d7c0-11eb-9371-447b1238cc7f.png)

I have set all the paths of the scripts like this

![image](https://user-images.githubusercontent.com/19240467/123564056-8e897d80-d7c0-11eb-817b-8a1ca013b0ba.png)

![image](https://user-images.githubusercontent.com/19240467/123564084-a7922e80-d7c0-11eb-9779-225e29af4bd5.png)

And this is the script error

![image](https://user-images.githubusercontent.com/19240467/123564097-b8db3b00-d7c0-11eb-9e83-0144cce2fdd0.png)

**Comment by snibgo** [comment]
I have responded on your other thread https://github.com/ImageMagick/ImageMagick/discussions/3860

**Comment by FurkanGozukara** [comment]
@dlemstra the results are simply amazing

Would you consider adding this to ImageMagick?

I had very hard time to make it work and even so it doesn't work when there are space in folder names

**Comment by snibgo** [comment]
The script is quite complex because of the options: upscale or downscale, with or without sharpening, and with or without deconvolution.

When it is used for downscaling with sharpening but no deconvolution, the script simplifies to this command (Windows BAT syntax):
```
magick ^
  input.png ^
  ( +clone -alpha extract
    -resize 100x100 -write mpr:ALP +delete ^
  ) ^
  -alpha off ^
  -colorspace RGB ^
  -define filter:c=0.1601886205085204 -filter Cubic -distort Resize 100x100 ^
  ( -clone 0 ^
    -gamma 3 ^
    -define convolve:scale=50%%,100 -morphology Convolve DoG:3,0,0.4981063336734057 ^
    -gamma 0.3333333333333333 ^
  ) ^
  ( -clone 0 -define convolve:scale=50%%,100 -morphology Convolve DoG:3,0,0.4981063336734057 ) ^
  -delete 0 ^
  ( -clone 1 -colorspace gray -auto-level ) ^
  -compose over -composite ^
  -set colorspace RGB -colorspace sRGB ^
  mpr:ALP -compose CopyOpacity -composite ^
  +depth ^
  output.png
```
Where both occurences of "100x100" are the usual IM resize argument, and both occurences of "50" (as in "-define convolve:scale=50%,100") are the amount of sharpening, typically 1 to 100 but can be larger eg 150 or 200.

**Reply by FurkanGozukara** [reply]
this can be added perhaps? I have used this command and it works awesome compared to unsharp

`call %PICTBAT%resampHM input.png dimensions d 150 . . output.png`

although due to pathing problem I had to copy every file into a non space having folder (file names as well so I copied file processed and copied again)

**Comment by snibgo** [comment]
Instead of running the script, you could use the command I show above.

**Reply by fmw42** [reply]
Just make a bat script of snibgo's downsample code and drop as many images onto it as you want.

**Reply by FurkanGozukara** [reply]
There are some static variables are they static? or changes according input such as

`convolve:scale=50%%,**100**`
`c=0.1601886205085204`
`DoG:3,0,0.4981063336734057`

so I need to input output and resize and which value is the sharpening?  

is 50%% static? what is this **100**?

I still prefer this to be added to the ImageMagick like `unsharp ` @dlemstra because I think it is a very good addition

@snibgo perhaps you can help them to add?

**Comment by snibgo** [comment]
In the command I show, "100x100" occurs twice. Change both of those to be whatever size you want.

Also in that command, the number "50" occurs twice. Change both of those to be whatever sharpness you want.

Those are the only variables. The other numbers are constant.

It could be implemented as an operation something like:
```
-sharp-shrink 100x100+50
```
